cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -325,9 +331,11 @@ impl ABIMachineSpec for AArch64MachineDeps {
                 std::cmp::max(size, 8)
             };
 
-            // Align the stack slot.
-            debug_assert!(size.is_power_of_two());
-            next_stack = align_to(next_stack, size);
+            if call_conv != isa::CallConv::Winch || args_or_rets != ArgsOrRets::Rets {
```

Comment:
Since we are checking this condition so frequently, it may help to define an `is_winch_return` boolean at the start of this procedure that we can reuse throughout.